### PR TITLE
Clarify breaking release frequency

### PR DESCRIPTION
**Pull request by nicoburns**
# Objective

Bevy's README is ambiguous about how often "frequent" breaking changes occur. And some people [have taken it](https://news.ycombinator.com/item?id=35045509) to mean that breaking changes may occur every few days in patch releases. I personally find this reading a little odd, but it never hurts to be clear.

## Solution

Be specific that breaking changes in Bevy tend to land on a roughly quarterly basis.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ Bevy is a refreshingly simple data-driven game engine built in Rust. It is free
 
 ## WARNING
 
-Bevy is still in the _very_ early stages of development. APIs can and will change (now is the time to make suggestions!). Important features are missing. Documentation is sparse. Please don't build any serious projects in Bevy unless you are prepared to be broken by API changes constantly.
+Bevy is still in the _very_ early stages of development. APIs can and will change (now is the time to make suggestions!). Important features are missing. Documentation is sparse. Please don't build any serious projects in Bevy unless you are prepared to be broken by API changes on a roughly quarterly basis.
```
Comment (IceSentry) [inline README.md:16]: The official train release schedule is every 3 months, so I would rephrase it with this in mind. https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule

Personally, I still think the user on hacker news assumed something that was never said. Bevy uses semver and nothing here claims otherwise but I guess it's true that being more accurate doesn't hurt this warning.

**Review by alice-i-cecile — APPROVED** [review]
I think this is clearer, and an uncontroversial improvement.

Very funny though: I made a note to make this exact PR too...

File: README.md (hunk 1 above)
Comment (nicoburns) [reply README.md:16]: That link says "approximately once every three months", this says "roughly quarterly". Those would be the same, right?

> Personally, I still think the user on hacker news assumed something that was never said

Agreed on this point

Comment (alice-i-cecile) [reply README.md:16]: Yep, those are the same period of time.

**Comment by alice-i-cecile** [comment]
We could add a note about how semver works in Rust?

**Comment by alice-i-cecile** [comment]
Ping @laundmo, since I saw you on that thread too.

File: README.md (hunk 1 above)
Comment (IceSentry) [reply README.md:16]: > Those would be the same, right?

Correct, what I meant is that I think this should explicitly mention the concept of train release and maybe even link it.

**Comment by laundmo** [comment]
i would prefer if it clarified more what the drawbacks are and is less "don't use this its really new" because bevy has already matured a lot.

maybe something along the lines of:

> Bevy is still in early development, and while we try our best you should be prepared for the following:
    - Important features missing
    - Sparse documentation
    - breaking releases roughly every 3 months (with a migration guide)
    Use at your own risk.

File: README.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Bevy is a refreshingly simple data-driven game engine built in Rust. It is free
 
 ## WARNING
 
-Bevy is still in the _very_ early stages of development. APIs can and will change (now is the time to make suggestions!). Important features are missing. Documentation is sparse. Please don't build any serious projects in Bevy unless you are prepared to be broken by API changes constantly.
+Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. And minor versions of Bevy containing breaking changes to the API are released [approximately once every 3 months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). Use at your own risk.
```
Comment (alice-i-cecile) [inline README.md:16]:
```suggestion
Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. A new version of Bevy containing serious breaking changes to the API is released [approximately once every 3 months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). Use at your own risk.
```

Comment (alice-i-cecile) [reply README.md:16]: They are technically minor versions, but I think that carries the wrong connotation.

**Comment by laundmo** [comment]
just for fun i asked bing chat what it would write:

> Bevy is a young project, and we are constantly adding new features and improving existing ones. However, this also means that you may encounter some limitations or challenges when using it, such as:
>   - Missing or incomplete functionality
>   - Sparse documentation that covers the basics but not the advanced topics
 >  - Breaking changes every 3 months when we release a new version of Bevy (with a migration guide to help you update your code) 
 
 >Please use Bevy only if you are willing to accept these challenges and risks.

**Review by IceSentry — APPROVED** [review]

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]

**Review by laundmo — APPROVED** [review]

**Comment by nicoburns** [comment]
> just for fun i asked bing chat what it would write:
> 
> > Bevy is a young project, and we are constantly adding new features and improving existing ones. However, this also means that you may encounter some limitations or challenges when using it, such as:
> > 
> > * Missing or incomplete functionality
> > * Sparse documentation that covers the basics but not the advanced topics
> > * Breaking changes every 3 months when we release a new version of Bevy (with a migration guide to help you update your code)
> 
> > Please use Bevy only if you are willing to accept these challenges and risks.

Huh, that's actually pretty good!

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@cart, you've wanted to vet these changes in the past, so I'm adding Controversial. Should be real quick to review though.

**Review by james7132 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ Bevy is a refreshingly simple data-driven game engine built in Rust. It is free
 
 ## WARNING
 
-Bevy is still in the _very_ early stages of development. APIs can and will change (now is the time to make suggestions!). Important features are missing. Documentation is sparse. Please don't build any serious projects in Bevy unless you are prepared to be broken by API changes constantly.
+Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. A new version of Bevy containing serious breaking changes to the API is released [approximately once every 3 months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). Use at your own risk.
```
Comment (cart) [inline README.md:16]:
```suggestion
Bevy is still in the early stages of development. Important features are missing. Documentation is sparse. A new version of Bevy containing breaking changes to the API is released [approximately once every 3 months](https://bevyengine.org/news/bevy-0-6/#the-train-release-schedule). We provide [migration guides](https://bevyengine.org/learn/book/migration-guides/), but we can't guarantee migrations will always be easy. Use only if you are willing to work in this environment.
```

"Use at your own risk" seems a bit to harsh and ominous to me. The kind of thing you sign on a waiver before doing something hazardous to your health. I swapped that out for something a bit less loaded. Also added a link to migration guides.

Comment (alice-i-cecile) [reply README.md:16]: I'm on board with that wording.
